express-notes-api/server.js

```diff
@@ -13,16 +13,19 @@ app.get('/api/notes', async (req, res) => {
   for (const note in data.notes) {
     array.push(data.notes[note]);
   }
-  res.status(200).send(array);
+  res.status(200).json(data);
 });
 
 // gets a single note by id
 app.get('/api/notes/:id', async (req, res) => {
   const data = await fetchFile();
   const id = req.params.id;
 
-  idErrorHandler(id, data, res);
-  res.status(200).send(data.notes[id]);
+  if (idErrorHandler(id, data, res)) {
+    idErrorHandler(id, data, res);
```

Comment:
You don't want to call it a second time. That will try to send a second response, which will cause an Express error.